packages/core/src/textures/TextureSystem.ts

```diff
@@ -1,13 +1,100 @@
 import { BaseTexture } from './BaseTexture';
 import { GLTexture } from './GLTexture';
 import { removeItems } from '@pixi/utils';
-import { MIPMAP_MODES, WRAP_MODES, SCALE_MODES, TYPES, SAMPLER_TYPES } from '@pixi/constants';
+import { FORMATS, MIPMAP_MODES, WRAP_MODES, SCALE_MODES, TYPES, SAMPLER_TYPES } from '@pixi/constants';
 
 import type { ISystem } from '../ISystem';
 import type { Texture } from './Texture';
 import type { IRenderingContext } from '../IRenderingContext';
 import type { Renderer } from '../Renderer';
 
+const glTypeFormatLookup: { [type: number]: { [format: number]: number } } = ('WebGL2RenderingContext' in self ? {
```

Comment:
in the spirit of not making our class too large, would be cool to move this to an external file!

Reply:
Should it be in @pixi/constants?

Reply:
No, its gl-related, so it should be in core. Making it the same way as in StateSystem will be fine.

Reply:
I fine with that - but @ivanpopelyshev everything in @pixi/constants _is_ gl-related.

Reply:
I moved it to the an external file similar to the `StateSystem` example.